backends-velox/src/main/scala/org/apache/spark/sql/execution/datasources/VeloxWriteQueue.scala

```diff
@@ -41,7 +45,9 @@ class VeloxWriteQueue(
   private val writeThread = new Thread(
     () => {
       try {
-        datasourceJniWrapper.write(dsHandle, scanner)
+        datasourceJniWrapper.write(
+          dsHandle,
+          new ColumnarBatchInIterator(new CloseableColumnBatchIterator(scanner).asJava))
       } catch {
         case e: Throwable =>
           writeException.set(e)
```

Comment:
@JkSelf I think it's OK to remove all the write queue stuffs in future, since Velox's Parquet writer exposes a "write(rv)" API. Previously in Gazelle we had to pass a iterator to Arrow library for writing so that a queue was required.

Reply:
Indeed, write performance might be impacted then. So some sort of PoCs will be needed.

Reply:
@zhztheplayer If each batch calls a JNI function here, it may impact performance. However, I cannot recall the specific performance data at the moment. We can conduct a few POCs tests later to gather more information. Additionally, we are working on upgrading to Spark 3.4. With this update, we can directly incorporate a WriteTransformer to achieve Parquet write functionality, eliminating the need for these unnecessary lines of code.